File: 문제풀이/chap3-4/prob3-4-4.c
#include <stdio.h>

int N;
double Answer = 0;

int main() {
	// 입력
	scanf("%d", &N);
	
	// 기댓값 구하기
	for (int i = N; i >= 1; i--) {
		Answer += 1.0 * N / i;
	}
	
	// 출력
	printf("%.12lf\n", Answer);
	return 0;
}

File: 문제풀이/chap3-4/prob3-4-4.java
import java.util.*;

class Main {
	public static void main(String[] args) {
		// 입력
		Scanner sc = new Scanner(System.in);
		int N = sc.nextInt();
		
		// 답 계산
		double Answer = 0.0;
		for (int i = N; i >= 1; i--) {
			Answer += 1.0 * N / i;
		}
		
		// 출력
		System.out.format("%.12f\n", Answer);
	}
}